### PR TITLE
Initial implementation Feature: Remote Buzzer Server

### LINKED ISSUE #93: Remote Buzzer Server
**Is your feature request related to a problem? Please describe.**
In a scenario where the Raspi based webserver serves photobooth over WLAN to a remote display screen (say a browser on a WLAN-connected tablet), the approach to trigger via a local access key does not work. Also, to trigger via a simulated local "[button press" script approach](https://github.com/andreknieriem/photobooth/wiki/FAQ#can-i-use-hardware-button-to-take-a-picture-on-my-raspberry-pi) based on the status of a GPIO pin does not work. 

**Describe the solution you'd like**
Be able to leverage any kind of remote trigger (i.e. buzzer, photo sensor, etc. - here the creativity starts) along with the Raspi. This remote trigger device could be connected via GPIO pin, Bluetooth, WLAN, etc. and trigger a photo / collage on the remote screen (tablet). Also to work for screens attached to the Raspi directly. 

**Proposal**
Implement a io_socket based client / server approach. The io server runs on the Raspberry PI. It will monitor GPIO Pins. It will also act as a server for any other trigger clients, e.g. it should be able to monitor BT connected trigger devices. 

I submitted pull request #94  and below please find a more detailed description of what I have implemented. If in general terms that feature is of interest to be added to the dev branch, I am happy to maintain and enhance as we go.

### PR DESCRIPTION
**Pull request by jacques42**
Initial implementation **Feature: Remote Buzzer Server** based on io sockets

Fix https://github.com/andi34/photobooth/issues/93

### CODE REVIEW

**Comment by andi34** [comment]
Thanks a lot! :)

Could you please take a look at the lint test fail?
https://github.com/andi34/photobooth/runs/968714117?check_suite_focus=true

**Comment by jacques42** [comment]
I've done some improvement work on the code and also the eslint sanity. However the Lint check still failes for reasons and on files that I have not touched. Not sure but I assume I have to rebase on the latest 2.8 branch and build on top. 

I am going to be away for the coming two weeks, so please bear with me until early September for to be able to come back to this.

**Comment by andi34** [comment]
Thanks and take your time :)
Maybe lint fails because of your changes to .eslintrc.json?

**Comment by jacques42** [comment]
I definitely appreciate if you would take a look. The [eslint rule that throws the errors](https://eslint.org/docs/rules/no-unused-vars) is `no-unused-vars` and is derived from the base rule set (`"extends": "eslint:recommended"`), which has been in the config file since.

```
/home/runner/work/photobooth/photobooth/src/js/adminshortcut.js
  8:10  error  'adminsettings' is defined but never used  no-unused-vars
/home/runner/work/photobooth/photobooth/src/js/chromakeying.js
  82:10  error  'setBackgroundImage' is defined but never used  no-unused-vars
/home/runner/work/photobooth/photobooth/src/js/photoinit.js
  3:10  error  'initPhotoSwipeFromDOM' is defined but never used  no-unused-vars
```
Changes I've done in the commits of this PR have not touched those three files, so I have no idea why this rule tiggers now. Of course I can remove the variables from the code to fix it. But in order to be better safe than sorry, I appreciate you looking at it as surely you have a better perspective on what's going on there.

The changes I've done to `.eslintrc.json` where for two reasons:

1. Error handler implementation
I am using a process error handler for `uncaughtExceptions`. If that triggers, the code does some clean-up and logging, and then the process must exit by all means, as something went wrong in a non-recoverable way. So I use the function `process.exit()` in that exception handler. The use of `process.exit()` triggered eslint rule `no-process-exit`. In the [rule documentation ](https://eslint.org/docs/rules/no-process-exit) it is marked as depreciated since eslint v7, with a recommendation to include the plugin `eslint-plugin-node`. So I did include this plug-in, added the `plugin:node/recommended` rule set and then turned off the depreciated rule `no-process-exit` in the `.eslintrc.json` configuration file.

2. Loading of Photobooth configuration 
At the start of the process, it reads the Photobooth current configuration. For that I decided to use a syncronous (blocking) function call `execSync`. Reason being is I'd like for this operation to complete before any further processing happens - simply to ensure operation completes first so that the Photobooth configuration data is available onwards. That synchronous function call triggered eslint rule `no-sync`. Similarly, the [rule documentation ](https://eslint.org/docs/rules/no-sync) says this rule is depreciated as of ESlint v7 and recommends to use the `eslint-plugin-node`. So I did accordingly, including to turn of rule `no-sync` in the `.eslintrc.json` configuration file.

Currently I can't see a good reason why that changes to `.eslintrc.json` could have caused to throw above three errors as new. But I don't know what I don't know. So I definitely appreciate your help looking into this. 

And of course, if there are better ways / better code to implement my items 1) and 2) I am super eager to learn. Please let me know. 

Talk to you in September !

**Comment by andi34** [comment]
Hey and thanks for taking a look at the lint errors.
Don't worry about the dev branch here being ahead of yours, we can rebase your commits once everything is setup.
There's a reason for "Revert "fix (slideshow): exit slideshow on close""?

**Comment by jacques42** [comment]
Removing it was a mistake, I added it back in.

File: install-raspbian.sh [hunk 1]
```diff
@@ -269,6 +269,16 @@ EOF
 
 fi
 
+info "### Enable Nodejs GPIO access - please reboot"
```
Comment (andi34) [inline install-raspbian.sh]: Maybe optional via request?

Comment (jacques42) [reply install-raspbian.sh]: The new overlay in /boot/config.txt will only activate at reboot. I did not find any way to activate this without reboot. And without this overlay, there is stability issues it seems.

Of course the reboot at this point is optional and not using the remote buzzer feature, the reboot is not required at all. 

I would not add a prompt here as right at the end of the script, anyways the user is prompted for a possible reboot.

**Pull request merged**